apis/kueue/v1beta2/multikueue_types.go

```diff
@@ -51,21 +51,41 @@ type KubeConfig struct {
 	// which the kueue controller manager is running. The config should be stored in the "kubeconfig" key.
 	// +kubebuilder:validation:MaxLength=256
 	// +kubebuilder:validation:MinLength=1
-	// +required
-	Location string `json:"location,omitempty"`
+	// +optional
+	Location string `json:"location,omitempty"` //nolint:kubeapilinter // should not be a pointer
 
 	// locationType of the KubeConfig.
 	//
 	// +kubebuilder:default=Secret
-	// +optional
 	// +kubebuilder:validation:Enum=Secret;Path
-	LocationType LocationType `json:"locationType"`
+	// +optional
+	LocationType LocationType `json:"locationType,omitempty"` //nolint:kubeapilinter // should not be a pointer
```

Comment:
Revert drive-by changes to the KubeConfig struct. Actually, Location remains required field for KubeConfig. It is just that KubeConfig is optional.